qiskit/transpiler/instruction_durations.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Durations of instructions, one of transpiler configurations."""
+import warnings
+from typing import Optional, List, Tuple, Union, Iterable
+
+from qiskit.circuit import Barrier, Delay
+from qiskit.circuit import Instruction, Qubit
+from qiskit.providers import BaseBackend
+from qiskit.transpiler.exceptions import TranspilerError
+
+InstructionDurationsType = List[Tuple[str, Optional[Iterable[int]], int]]
```

Comment:
Shouldn't it be this?
```suggestion
InstructionDurationsType = Union[List[Tuple[str, Optional[Iterable[int]], int]],
                                 InstructionDurations]
```
I guess that isn't helpful for line 32, but it's confusing for an `InstructionDurations` instance to not satisfy `InstructionDurationsType` for usage elsewhere in the code

Reply:
839d727